src/main/daml/Daml/Finance/Instrument/Swap/Fpml/Instrument.daml

```diff
@@ -63,15 +63,23 @@ template Instrument
       getClaims Claim.GetClaims{actor} = do
         -- get the initial claims tree (as of the swap's acquisition time)
         let
-          calculateClaimsFromSwapStream s = do
+          getNotionalFromOtherSwapStream ref = do
```

Comment:
You don't really need the `do` block for the `Optional` monad, as you don't effectively make use of it.

Same for `calculateClaimsFromSwapStream` below.

Reply:
On the other hand, using for instance the `Either` monad would allow you to make sone nice error handling

Reply:
> Same for `calculateClaimsFromSwapStream` below.

I have changed it for `getNotionalFromOtherSwapStream`, but in `calculateClaimsFromSwapStream` I use `assertMsg`. Does that work without do? What is the benefit of modifying it?

Reply:
Right, `assertMsg` might not work without `do` as it outputs a monad.

The benefit of modifying it is (just in my opinion) cleaner code. For instance, If i see I `do` then I tend to check within which monad we are (often an `Update` but not always).